test_gpios.py

```diff
@@ -20,9 +20,26 @@ async def asyncSetUp(self):
         self.robot = await RobotClient.at_address(conf.address, opts)
         board = Board.from_robot(self.robot, "board")
         self.input_pin = await board.gpio_pin_by_name(conf.INPUT_PIN)
-        self.interrupt = await board.digital_interrupt_by_name(conf.INPUT_PIN)
         self.output_pin = await board.gpio_pin_by_name(conf.OUTPUT_PIN)
 
+        # Most boards have combination GPIO/PWM/interrupt pins. However, rarely
+        # they are separated to different pins (e.g., the Beaglebone AI-64 does
+        # not have GPIO functionality on the PWM pins or vice versa). If you
+        # define them separately, define these extra variables in the config.
+        # Otherwise, we'll reuse the same inputs and outputs.
+        try:
+            INTERRUPT_PIN = conf.INTERRUPT_PIN
+        except AttributeError:
+            INTERRUPT_PIN = conf.INPUT_PIN
```

Comment:
There are two pairs of pins right now: GPIO (output) pairing with GPIO (input), and PWM (output) pairing with interrupt (input). On non-Beaglebone boards, it's the same pair repeated twice.

As currently written, if you don't have an interrupt pin, the PWM tests will fail because we don't have a way to count whether they're accurate. If you don't have a GPIO output pin, the GPIO tests will fail (for obvious reasons). and if you don't have a PWM output pin, frankly you can use software PWM on a GPIO pin and it'll be okay.

Reply:
...I might have misunderstood your question. If you don't define `conf.INTERRUPT_PIN`, we reuse `conf.INPUT_PIN` and expect that the pin can do both GPIO input and digital interrupts. Similarly, if you don't define `conf.PWM_PIN`, we reuse `conf.OUTPUT_PIN` (which should always work: either it has hardware PWM and it works, or it doesn't and we fall back to software PWM).